src/net/tls.cc

```diff
@@ -1360,23 +1360,27 @@ class session : public enable_lw_shared_from_this<session> {
             gnutls_transport_set_errno(*this, EAGAIN);
             return -1;
         }
-        if (_output_pending.failed()) {
-            // copy exception for on_internal_error
-            auto ex = _output_pending.get_exception();
-            _output_pending = make_exception_future<>(ex);
-            on_internal_error(seastar_logger, format("tls: session: vec_push: unhandled pending output error: {}", ex));
-        }
         try {
+            ssize_t n; // Set on the good path and unused on the bad path
+            bool prev_output_pending_failed = _output_pending.failed();
+
+          // FIXME indentation
+          if (!prev_output_pending_failed) {
             scattered_message<char> msg;
             for (int i = 0; i < iovcnt; ++i) {
                 msg.append(std::string_view(reinterpret_cast<const char *>(iov[i].iov_base), iov[i].iov_len));
             }
-            auto n = msg.size();
+            n = msg.size();
             _output_pending = _out.put(std::move(msg).release());
+          }
             if (_output_pending.failed()) {
                 // exception is copied back into _output_pending
                 // by the catch handlers below
-                std::rethrow_exception(_output_pending.get_exception());
+                auto ex = _output_pending.get_exception();
+                if (prev_output_pending_failed) {
+                    seastar_logger.warn("tls: vec_push: pending output failed: {}", ex);
```

Comment:
Isn't this user-triggerable? A user can connect to a server and spam the logs this way.

Since we rethrow the exception, the caller can decide whether to log the error or not.

Reply:
How?
It's printed only in the rare case of an asynchromous error that isn't caught by wait_for_output.  It's actually quite hard to reproduce, even with error injection.

Reply:
I don't see that it merits a warning. What can the user do about it? report it to the application author?

I think it's okay to collapse two errors into one (as I think the code is doing) without additional noise in the logs. Swallowing errors is bad, but this isn't the case here.

Reply:
Ok, the audience for this error is us indeed, not the user, who can't do anything about it.